packages/dx-grid-core/src/plugins/filtering-state/computeds.ts

```diff
@@ -0,0 +1,16 @@
+import { FilterExpression, Filter } from '../../types';
+import { PureComputed } from '@devexpress/dx-core';
+
+export const filterExpression: PureComputed<
+  [Filter[], FilterExpression], FilterExpression
+> = (filters, expression) => {
+  // tslint:disable-next-line: no-object-literal-type-assertion
+  const selfFilterExpr = { filters, operator: 'and' as 'and' } as FilterExpression;
+  if (!expression) {
+    return selfFilterExpr;
+  }
+  return {
+    operator: 'and' as 'and',
```

Comment:
Are this `as` and the one on the next line are required? (in a simplified sandbox playground they not)

Reply:
Compiler throws an exception - it expects an 'and | or' type, but receives 'and' as string

Reply:
OK.